### PR TITLE
Pull request from Revitalize branch + fix for custom sketeton tool crash

### PR DESCRIPTION
**Pull request by ksuprynowicz**

### CODE REVIEW

File: metaverse_tools/armature/__init__.py [hunk 1]
```diff
@@ -28,9 +28,11 @@ def get_type_from_armature(obj):
         # TODO: Better Detection Methods Later using skeletons.
 
         if obj.data.bones.find("Chest"):
+            print("VRC Avatar Detected")
             return SkeletonTypes.VRC
 
         if obj.data.bones.find("Spine2") or obj.data.bones.find("HeadTop_End"):
+            print("Hifi Avatar Detected")
```
Comment (digisomni) [inline metaverse_tools/armature/__init__.py]:
```suggestion
            print("Vircadia Avatar Detected")
```

File: metaverse_tools/__init__.py [hunk 2]
```diff
@@ -394,57 +97,38 @@ def menu_func_import(self, context):
 
 def menu_func_export(self, context):
     
-    self.layout.operator(EXPORT_OT_MVT_TOOLSET_FBX.bl_idname, text="Hifi FBX (.fbx)")
-    self.layout.operator(EXPORT_OT_MVT_TOOLSET_FST_Writer_Operator.bl_idname,
-                         text="HiFi Avatar FST (.fst)")
-   # self.layout.operator(EXPORT_OT_MVT_TOOLSET_Export_GLTF_JSON.bl_idname,
+    self.layout.operator(EXPORT_OT_MVT_TOOLSET_FBX.bl_idname, text="Tivoli/Vircadia FBX (.fbx)")
```
Comment (digisomni) [inline metaverse_tools/__init__.py]:
```suggestion
    self.layout.operator(EXPORT_OT_MVT_TOOLSET_FBX.bl_idname, text="Vircadia FBX (.fbx)")
```

File: metaverse_tools/__init__.py [hunk 3]
```diff
@@ -394,57 +97,38 @@ def menu_func_import(self, context):
...
+    self.layout.operator(EXPORT_OT_MVT_TOOLSET_Hifi_FST_Writer_Operator.bl_idname,
+                         text="Tivoli/Vircadia Avatar FST (.fst)")
```
Comment (digisomni) [inline metaverse_tools/__init__.py]:
```suggestion
                         text="Vircadia Avatar FST (.fst)")
```

File: metaverse_tools/__init__.py [hunk 4]
```diff
@@ -394,57 +97,38 @@ def menu_func_import(self, context):
...
-    def execute(self, context):
-        url = "https://forums.highfidelity.com/t/high-fidelity-blender-add-on-version-1-1-10-released/13717"
-        if "windows-default" in webbrowser._browsers:
-            webbrowser.get("windows-default").open(url)
-        else:
-            webbrowser.open(url)
-
-        return {'FINISHED'}
+    self.layout.operator(EXPORT_OT_MVT_TOOLSET_Writer_FBX_JSON.bl_idname,
+                         text="Tivoli/Vircadia Metaverse Scene JSON / FBX (.json/.fbx)")
```
Comment (digisomni) [inline metaverse_tools/__init__.py]:
```suggestion
                         text="Vircadia Metaverse Scene JSON / FBX (.json/.fbx)")
```

File: metaverse_tools/files/fst/operator.py [hunk 5]
```diff
@@ -156,7 +156,7 @@ def draw(self, context):
         row.label(text="Avatar Export Successful.")
 
 
-class EXPORT_OT_MVT_TOOLSET_FST_Writer_Operator(bpy.types.Operator, ExportHelper):
+class EXPORT_OT_MVT_TOOLSET_Hifi_FST_Writer_Operator(bpy.types.Operator, ExportHelper):
     """ This Operator is export a HighFidelity compatible FST and FBX of the current avatar.
```
Comment (digisomni) [inline metaverse_tools/files/fst/operator.py]:
```suggestion
    """ This Operator exports a Vircadia compatible FST and FBX of the current avatar.
```

File: metaverse_tools/files/fst/operator.py [hunk 6]
```diff
@@ -184,66 +184,19 @@ class EXPORT_OT_MVT_TOOLSET_FST_Writer_Operator(bpy.types.Operator, ExportHelper
     embed: BoolProperty(default=False, name="Embed Textures",
                          description="Embed Textures to Exported Model. Turn this off if you are having issues of Textures  not showing correctly in elsewhere.")
```
Comment (digisomni) [inline metaverse_tools/files/fst/operator.py]:
```suggestion
                         description="Embed Textures to Exported Model. Turn this off if you are having issues of Textures not showing correctly in elsewhere.")
```

File: metaverse_tools/.vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "C:\\Users\\Malactus\\AppData\\Local\\Programs\\Python\\Python39\\python.exe"
```
Comment (digisomni) [inline metaverse_tools/.vscode/settings.json]: This file should be added to the .gitignore probably.

File: metaverse_tools/ui/armature_tools.py [hunk 8]
```diff
@@ -108,6 +105,53 @@ def execute(self, context):
...
+    bl_region_type = "TOOLS"
+    bl_space_type = "VIEW_3D"
+    bl_options = {'REGISTER', 'UNDO', 'INTERNAL'}
+
+    target_name: bpy.props.StringProperty()
+    different_name: bpy.props.BoolProperty(default=False)
+
+    @classmethod
+    def poll(self, context):
+        return context.selected_bones is not None and len(context.selected_bones) >= 1
```
Comment (digisomni) [inline metaverse_tools/ui/armature_tools.py:123]: Just making sure this is a correct expression. :)

File: metaverse_tools/ui/modules/hifi.py [hunk 9]
```diff
@@ -28,7 +28,7 @@
 category = "MVT: HiFi Tools"
 
 
-class AVATAR_PT_MVT_metaverse_toolsET(bpy.types.Panel):
+class AVATAR_PT_MVT_metaverse_toolset(bpy.types.Panel):
     """ Panel for HF Avatar related conversion tools """
     bl_label = "Hifi Avatar Tools"
```
Comment (digisomni) [inline metaverse_tools/ui/modules/hifi.py]:
```suggestion
    """ Panel for Vircadia Avatar related conversion tools """
    bl_label = "Vircadia Avatar Tools"
```

File: metaverse_tools/utils/bones/custom.py [hunk 10]
```diff
@@ -547,25 +550,152 @@ def draw(self, context):
 
         # After selecting armature, iterate through bones.
 
+
+
+class AVATAR_OT_MVT_TOOLSET_Avatar_Rebinder_Operator(bpy.types.Operator):
+    """ Avatar Binding Tool allowing one to attempt convert a existing custom skeleton with mesh into a user defined skeleton
```
Comment (digisomni) [inline metaverse_tools/utils/bones/custom.py]:
```suggestion
    """ Avatar Binding Tool allows one to attempt convert a existing custom skeleton with mesh into a user defined skeleton
```

File: metaverse_tools/utils/bones/makehuman.py [hunk 11]
```diff
@@ -196,7 +196,7 @@ def create_blink_shapes(shapekey, armature1, armature2):
         bpy.ops.object.mode_set(mode='OBJECT')
         bpy.ops.object.select_all(action='DESELECT')
         obj.select_set(state=True)
-        bpy.ops.object.modifier_apply(apply_as='SHAPE', modifier='ARMATURE')
+        bpy.ops.object.modifier_apply( modifier='ARMATURE')
```
Comment (digisomni) [inline metaverse_tools/utils/bones/makehuman.py]:
```suggestion
        bpy.ops.object.modifier_apply(modifier='ARMATURE')
```

**Review by digisomni — CHANGES_REQUESTED** [review]

File: metaverse_tools/.vscode/settings.json (hunk 7 above)
Comment (ksuprynowicz) [reply metaverse_tools/.vscode/settings.json]: Sure!
Fixed it in https://github.com/vircadia/Blender-Metaverse-Addon/pull/1/commits/b858ad1c0a74555dd85a457c84e5b91d830e9122

File: metaverse_tools/ui/armature_tools.py (hunk 8 above)
Comment (ksuprynowicz) [reply metaverse_tools/ui/armature_tools.py:123]: I didn't look at this file yet, I will test if it works

File: .vscode/settings.json [hunk 12]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.pythonPath": "C:\\Users\\Malactus\\AppData\\Local\\Programs\\Python\\Python39\\python.exe",
```
Comment (digisomni) [inline .vscode/settings.json]:
```suggestion
```

Comment (ksuprynowicz) [reply .vscode/settings.json]: Fixed in https://github.com/vircadia/Blender-Metaverse-Addon/pull/1/commits/283a74edae6ef56ef9f6bfa87314a17c31df98be

**Pull request merged**